defaults/config.js

```diff
@@ -35,6 +35,14 @@ module.exports = {
 	// This value is set to `9000` by default.
 	port: 9000,
 
+	// ### `socketPermissions`
+	//
+	// Set the permissions of the UNIX domain socket.
+	// Only applies if the `host` option is a UNIX domain socket.
+	//
+	// This value is set to `0o660` by default.
+	socketPermissions: 0o660,
```

Comment:
Not sure what FreeBSD does, but on Linux this isn't actually enough to set the permissions up.
Chances are that you want to restrict access to a certain group, meaning you need a way to specify owner / group as well as the permission bits.

Meaning, we should probably make this an object no?
I'm fine with just merging it with the current ability to set permission bits, but should someone want user / group in the future it should be extensible without introducing yet another top level option.

Meaning, we could have
```js
socketPermissions: {
    mode: 0o660,
    user: undefined,
    group: undefined
}
```
eventually, starting with only mode.

Does this make sense or am I completely of the right track here?

Reply:
Relevant section:

> Pathname socket ownership and permissions
       In the Linux implementation, pathname sockets honor the permissions of the directory they are in.  Creation of a new socket fails if the process does not have write and search (execute) permission on the directory in which the
       socket is created.
>
> On  Linux,  connecting to a stream socket object requires write permission on that socket; sending a datagram to a datagram socket likewise requires write permission on that socket.  POSIX does not make any statement about the
       effect of the permissions on a socket file, and on some systems (e.g., older BSDs), the socket permissions are ignored.  Portable programs should not rely on this feature for security.
>
>When creating a new socket, the owner and group of the socket file are set according to the usual rules.  The socket file has all permissions enabled, other than those that are turned off by the process umask(2).  
>
>The owner, group, and permissions of a pathname socket can be changed (using chown(2) and chmod(2)).

Reply:
Looking into different projects, I only found gogs/gitea, that have a UNIX_SOCKET_PERMISSION option https://docs.gitea.io/en-us/config-cheat-sheet/ or https://github.com/gogs/gogs/blob/main/conf/app.ini.